tests/console/validate_modify_existing_partition.pm

```diff
@@ -23,12 +23,21 @@ sub run {
     select_console "root-console";
 
     foreach my $part (keys %$test_data) {
-        record_info("Check $test_data->{fs_type}", "Verify that the partition filesystem is $test_data->{$part}->{fs_type}");
-        my $fstype = script_output("df -PT $test_data->{$part}->{mount_point} | grep -v \"Filesystem\" | awk '{print \$2}'");
+        record_info("Check $test_data->{$part}->{fs_type}", "Verify that the partition filesystem is $test_data->{$part}->{fs_type}");
+
+        my $fstype;
+        # We cannot use df for swap: it is not actually mounted but activated
+        if ("$test_data->{$part}->{fs_type}" eq "swap") {
+            $fstype = script_output("lsblk | grep $test_data->{$part}->{existing_partition} | sed 's/.* //;s/[][]//g' | tr '[:upper:]' '[:lower:]'");
```

Comment:
It would be more readable if you use same approach than line below with `awk` and avoid `tr` converting lower case in Perl with `lc`.

Reply:
wait, the brackets, I forgot that, not sure, I guess some regex would be more simple, or even include it in the test_data, another option it would be to use other command that does not return that with brackets.

Reply:
Initially I had found a complicated sed regex to do everything at once, but openqa was not digesting it, even with more \ and '". I can try to replace tr with lc. Or, another solution is to write [SWAP] in the yaml, as Sofia did

Reply:
> Or, another solution is to write [SWAP] in the yaml, as Sofia did
Actually, no, because then if want to use that code to resize swap, it will be a problem. I have not found better than lsblk, df or mount cannot be used, free could do the trick, but we'd still need some transformation like` if (script_output(free | grep Swap | awk '{print '2}')) { $fstype = "swap"} 
`

Reply:
For example, this is shorter and it could avoid the `if ` condition:
`blkid | grep $test_data->{$part}->{existing_partition} | awk -F "\"" '{print $4}'`.
For this case does not make sense to expend too much on that, but I guess if blkid is good enough for the case, in the future we could parse it in Perl, basically we can process it line by line, then split fields by spaces and again split fields by `=` and remove `"` from values, giving us at the end a hash with key values easy to check and compare with the hash in the expectation.

Reply:
blkid does not tell us if the swap is active, it just tells there's a swap partition on the disk

Reply:
You can combine same approach: lsblk | grep <device> | awk '{print $7}'` and then:
```
$str =~ s/([\[\]])//g;
lc $str
```
It is more readable but it adds two extra lines of code (I don't know how to put that in one line).
Feel free to leave it as you have it.

Reply:
thanks for the clarification about blkid, I didn't know that :)